test/TestCaseReader.cpp

```diff
@@ -137,6 +144,54 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSourcePath = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSourcePath;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSourcePath.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSourcePath.substr(0, remappingPos));
+					externalSourcePath = boost::trim_copy(externalSourcePath.substr(remappingPos + 1));
+				}
+
+				string externalSourceContent;
+				fs::path parent_path(m_filename);
+				parent_path = parent_path.normalize().remove_filename();
```

Comment:
`remove_filename()` modifies the path in place. I'd do either
```suggestion
				fs::path parent_path = fs::path(m_filename).normalize();
				parent_path.remove_filename();
```
or
```suggestion
				fs::path parent_path = fs::path(m_filename).normalize().parent_path();
```

By the way, where does `normalize()` come from? I do not see it in the [docs for `boost::filesystem` in Boost 1.75.0](https://www.boost.org/doc/libs/1_75_0/libs/filesystem/doc/reference.html) and I saw comments mentioning that it has been removed from boost. That's weird because the tests pass in this PR.

Reply:
Good catch! Looks like `normalize()` got deprecated. I use `canonical()` now.